pyccel/ast/literals.py

```diff
@@ -33,44 +33,64 @@ class Literal(PyccelAstNode):
 class LiteralTrue(sp_BooleanTrue, Literal):
     """Represents the python value True"""
     _dtype     = NativeBool()
-    _precision = default_precision['bool']
+    def __init__(self, precision = default_precision['bool']):
+        self._precision = precision
+
+    def set_precision(self, precision):
+        self._precision = precision
 
 #------------------------------------------------------------------------------
 class LiteralFalse(sp_BooleanFalse, Literal):
     """Represents the python value False"""
     _dtype     = NativeBool()
-    _precision = default_precision['bool']
+    def __init__(self,precision = default_precision['bool']):
+        self._precision = precision
+
+    def set_precision(self, precision):
+        self._precision = precision
 
 #------------------------------------------------------------------------------
 class LiteralInteger(sp_Integer, Literal):
     """Represents an integer literal in python"""
     _dtype     = NativeInteger()
-    _precision = default_precision['int']
+    _precision = default_precision['integer']
     def __new__(cls, val):
         ival = int(val)
         obj = Expr.__new__(cls, ival)
         obj.p = ival
         return obj
+    def __init__(self, value, *, precision = default_precision['integer']):
+        self._precision = precision
+
+    def set_precision(self, precision):
+        self._precision = precision
 
 #------------------------------------------------------------------------------
 class LiteralFloat(sp_Float, Literal):
     """Represents a float literal in python"""
     _dtype     = NativeReal()
-    _precision = default_precision['real']
+    def __init__(self, value, *, precision = default_precision['float']):
```

Comment:
Can you explain why you changed the default precision ( I assume these are dictionary keys have equivalent values?)